### PR TITLE
Generalize utility functions previously used for a single purpose

### PR DESCRIPTION
**Pull request by Equilibris**
Make `mkConstructors` and `getExpectedType` more versatile, this is done by extraction constants and turning them into arguments. This is done so they can be reused later at other points where a similar functionality is wanted.

This should in practice be non-functional in the greater context of the code, but it allows for these functions that can be hooked into for more fine-grained control without requiring more boilerplate code.

### CODE REVIEW

**Comment by alexkeizer** [comment]
Please give this PR a more descriptive name, and write PR description. Also, when stacking PRs it's best to keep this one in draft mode, and only request a review here after the other one gets merged, to avoid double reviewing the same code

File: Qpf/Macro/Data/Constructors.lean
Comment (Equilibris) [inline Qpf/Macro/Data/Constructors.lean:4]:
```suggestion
open Lean Meta Elab Elab.Command
```

Comment (Equilibris) [inline Qpf/Macro/Data/Constructors.lean:4]: I will fix this tgt with other things found in review

**Review by AtticusKuhn — APPROVED** [review]
Looks good to me. I like how this PR is small and atomic.

**Review by tobiasgrosser — APPROVED** [review]
Can you please explain in the commit message in which way these functions have been made more versatile.

Otherwise, LGTM.

File: Qpf/Macro/Data/Constructors.lean
Comment (alexkeizer) [inline Qpf/Macro/Data/Constructors.lean:16]: I'd prefer the return type (presumably `CommandElabM Unit`?) to be explicit.
Also, please write a docstring to explain what this docs (and crucially, how it differs from `mkConstructors`).

Finally, I like to limit lines to 100 chars wide, just like in Mathlib. This line feels like it might be too long (but I haven't checked, we really should set up a linter of some sort)

Comment (alexkeizer) [inline Qpf/Macro/Data/Constructors.lean:16]:
```suggestion
def mkConstructorsWithNameAndType (view : DataView) (shape : Name) (nameGen : CtorView → Name) (ty : Term) := do
```
I feel like having the argument generate a `Name` is cleaner than making it generate `Ident`s

File: Qpf/Macro/Data/View.lean
Comment (alexkeizer) [inline Qpf/Macro/Data/View.lean:125]: Please explain in the docstring what `id` does, also, here, too I feel `Name` is more appropriate
```suggestion
def DataView.getExpectedTypeWithId (view : DataView) (id : Name) : Term
```

File: Qpf/Macro/Data/View.lean
Comment (alexkeizer) [inline Qpf/Macro/Data/View.lean:132]: The style is to have `:=` on the same line as the type
```suggestion
def DataView.getExpectedType (view : DataView) : Term := 
  view.getExpectedTypeWithId (mkIdent view.shortDeclName)
```

**Review by alexkeizer — APPROVED** [review]
LGTM, feel free to merge after addressing the comments (either by accepting the suggestion, or leaving a comment to justify why you think the original thing is better; none of my comments I would consider super blocking)

File: Qpf/Macro/Data/View.lean
Comment (alexkeizer) [inline Qpf/Macro/Data/View.lean:136]: This function also has the wrong style, the `:=` should be on the same line as `CommandElabM Term`

(This is code you haven't touched in this PR, though, so feel free to ignore.

Also, I don't remember if I wrote this, but it is very possible. I wrote a bunch of this stuff before I knew what the style should've been, so it's not necessarily good to just mimic the style of what's already there. Sorry 'bout that)

**Pull request merged**
